### PR TITLE
Jest Snapshots and Doc Generation

### PR DESCRIPTION
**Pull request by bhough**
- Implements an example of snapshot testing with Jest.
- Implements automatic docs generation using JSDOCs .
- Adds watch task that both generates bundle and documentation.

### CODE REVIEW

**Comment by bhough** [comment]
@mxstbr this is ready for review

File: .eslintrc [hunk 1]
```diff
@@ -1,12 +1,24 @@
 {
-  "extends": "airbnb-base/legacy",
```
Comment (mxstbr) [inline .eslintrc]: Why was this removed? We want those linting rules?

File: src/helpers/StripUnit.js [hunk 2]
```diff
@@ -1,14 +1,26 @@
...
+import isInteger from '../internalHelpers/_isInteger'
+
 /**
- * Strips the unit from a given value.
- * @param {string} value - a string that is a number plus a unit of measure.
- * @return {number} - an integer without the unit.
+ * CSS to represent truncated text with an ellipsis.
+ * @static
+ * @param {string} [width = 100%] - Max-width for text to respect before being truncated.
+ * @return {Object} CSS as JS for truncating text.
```
Comment (mxstbr) [inline src/helpers/StripUnit.js]: These descriptions are all wrong I think? This should be "Strip unit of a given CSS value", "[value]" and it returns "Value without the unit"

**Comment by mxstbr** [comment]
This is AMAZING! 🎉 Love it!

Two small questions, then we're good to go here!

File: .eslintrc (hunk 1 above)
Comment (bhough) [inline .eslintrc]: I actually removed this because it only covered ES5 standards. I'd like to replace it with the full base library (which covers ES6+) if you are fine with that.

File: src/helpers/StripUnit.js (hunk 2 above)
Comment (bhough) [inline src/helpers/StripUnit.js]: Good catch. This is just an artifact of me testing the doc generator across multiple items.  I'll add proper documentation for this.

File: .eslintrc (hunk 1 above)
Comment (bhough) [inline .eslintrc]: We may want to add some overrides to their rules for our application as well. A couple examples:

- it throws errors for unnecessary quoted property names. Do we want to consistently quote them since there are so many hyphenated properties in css?

- It throws errors for missing spaces before function parentheses, but it looks like in `styled-components` we aren't enforcing that.

Comment (mxstbr) [inline .eslintrc]: Yeah just use `eslint-config-airbnb`, the standard one.

Yeah the consistent quoted property names sounds like a good idea, and I don't like spaces before function parentheses so whichever way you want to go there is fine with me!

**Comment by bhough** [comment]
alright @mxstbr take 2 :)

**Comment by mxstbr** [comment]
Amazing work!
